closed/src/java.base/share/classes/sun/security/rsa/NativeRSACore.java

```diff
@@ -138,17 +140,40 @@ public static int getByteLength(BigInteger b) {
             outputLen = getByteLength(n);
             output = new byte[outputLen];
             int verifyInt = verify ? outputLen : -1;
-            outputLen = nativeCrypto.RSADP(msg, msg.length, output, verifyInt, nativePtr);
+            RSADPBigNumRPointer rsadpBn = rsadpBnPtr.get();
+            if (rsadpBn == null) {
+                rsadpBn = new RSADPBigNumRPointer();
+                rsadpBnPtr.set(rsadpBn);
+            }
+            outputLen = nativeCrypto.RSADP(msg, msg.length, output, verifyInt, nativePtr, rsadpBn.getBN());
         } finally {
             key.returnNativePtr(nativePtr);
         }
-
         if (outputLen == -1) {
-            return null;
-        } else if (outputLen == -2) {
             throw new BadPaddingException("RSA private key operation failed");
         }
 
         return output;
     }
+
+    static class RSADPBigNumRPointer {
+        long rsadpBN = 0;
+
+        RSADPBigNumRPointer() {
+            this.rsadpBN = nativeCrypto.create_rsadp_BN();
+        }
+
+        @Override
+        protected synchronized void finalize() throws Throwable {
+            try {
+                nativeCrypto.free_rsadp_BN(rsadpBN);
+            } finally {
+                super.finalize();
```

Comment:
This is unnecessary, it doesn't do anything.

Reply:
That's right, I removed it.